### PR TITLE
Fix update reference for COPY_IN objects

### PR DESCRIPTION
**Pull request by gigiblender**
#### Description

This PR contains the following changes:
- Clean the Local/GlobalObjectState objects. Remove the concept of `owner` device.
- Fix TaskSchedule::updateReference for objects that are copied via a COPY_IN bytecode. `Resize::testUpdateReferenceCopyIn` tests this scenario.
- Expand profiling information for sync operations and differentiate between dispatch kernel and data transfer times. `TestProfiler::testProfilerEnabled` includes the sync profile info case.

#### Problem description

The problems fixed in this PR are the following:
1) When `TaskSchedule::updateReference` is called on a variable that is copied in (via a `COPY_IN` bytecode), the runtime thinks that the data is already on the device and will not perform any update. This is fixed by invalidating the object state and updating the necessary datastructures. (Check `TornadoExecutionContext::replaceObjectState`)

2) Expand the profiling information to include times for `sync` operations; Include call stack writing in the reported COPY_IN times. Split `DISPATCH_TIME` into `TOTAL_DISPATCH_DATA_TRANSFERS_TIME` and `TOTAL_DISPATCH_KERNEL_TIME`.

3) Remove the GridTask from the `TornadoVM` constructor. Instead, update the `GridTask` of the `TornadoVM` object separately, since the same `TornadoVM` object might be used with different `GridTask`s

#### Backend/s tested

- [X] OpenCL
- [X] PTX

#### OS tested

- [X] Linux
- [ ] OSx
- [ ] Windows

#### Did you check on FPGAs?

If possible, check your changes on FPGAs.

- [ ] Yes
- [X] No

#### How to test the new patch?

Running the unit tests should cover this PR. 
For the update reference fix, check `Resize::testUpdateReferenceCopyIn`

### CODE REVIEW

**Comment by jjfumero** [comment]
Please, update the PR description with the new unittests introduced in this patch.

File: runtime/src/main/java/uk/ac/manchester/tornado/runtime/profiler/TimeProfiler.java [hunk 1]
```diff
@@ -32,6 +32,8 @@
 
 public class TimeProfiler implements TornadoProfiler {
 
+    public static String NO_TASK_NAME = "noTask";
```
Comment (jjfumero) [inline runtime/src/main/java/uk/ac/manchester/tornado/runtime/profiler/TimeProfiler.java:35]: Add Java-doc why this taskName is neded. Also, you can relate to the open issue we have to discuss this.

**Review by jjfumero — CHANGES_REQUESTED** [review]

**Review by stratika — CHANGES_REQUESTED** [review]
The only part that I would suggest to update is the comment in `TornadoTaskSchedule.java` in order to be more clear.

File: drivers/ptx/src/main/java/uk/ac/manchester/tornado/drivers/ptx/PTXDeviceContext.java [hunk 2]
```diff
@@ -219,25 +219,42 @@ public int enqueueKernelLaunch(PTXModule module, CallStack stack, long batchThre
             blockDimension = scheduler.calculateBlockDimension(module);
             gridDimension = scheduler.calculateGridDimension(module, blockDimension);
         }
-        int kernelLaunchEvent = stream.enqueueKernelLaunch(module, writePTXStackOnDevice((PTXCallStack) stack), gridDimension, blockDimension);
+        int kernelLaunchEvent = stream.enqueueKernelLaunch(module, writePTXStackOnDevice((PTXCallStack) stack, module.metaData), gridDimension, blockDimension);
```
[the remark this answers is not in the document]
Comment (gigiblender) [reply drivers/ptx/src/main/java/uk/ac/manchester/tornado/drivers/ptx/PTXDeviceContext.java:222]: Yes, you're right. I should get a merge conflict once we merge that

**Review by stratika — APPROVED** [review]

**Pull request merged**